EIPS/eip-3074.md

```diff
@@ -27,125 +27,103 @@ While it is possible to emulate sponsored transactions (ex. [Gas Station Network
 
- - **`top - N`** - the `N`th most recently pushed value on the EVM stack, where `top - 0` is the most recent.
- - **`||`** - byte concatenation operator.
+### `AUTHORIZE` (`0xf8`)
 
-### Constants
+A new opcode `AUTHORIZE` is created at `0xf8`. It takes in 4 stack elements and returns 1 stack element.
 
-|     Constant     | Value  | Description                                                                         |
-| ---------------- | ------ |:----------------------------------------------------------------------------------- |
-| `SPONSORED_TYPE` | `0x03` | [EIP-2718](./eip-2718.md) transaction type reserved for transaction-like packages.  |
+#### Input
 
-### API
+| Stack      | Value        |
+| ---------- | ------------ |
+| `top - 0`  | `commit`     |
+| `top - 1`  | `yParity`    |
+| `top - 2`  | `r`          |
+| `top - 3`  | `s`          |
 
-#### Inputs
+#### Output
 
-`CALLFROM` must require the following stack arguments:
+| Stack      | Value               |
+| ---------- | ------------------- |
+| `top - 0`  | `authorizedAccount` |
 
-| Stack      | Value        |
-| ---------- | ------------ |
-| `top - 0`  | `yParity`    |
-| `top - 1`  | `r`          |
-| `top - 2`  | `s`          |
-| `top - 3`  | `sponsee`    |
-| `top - 4`  | `extra`      |
-| `top - 5`  | `gas`        |
-| `top - 6`  | `callee`     |
-| `top - 7`  | `value`      |
-| `top - 8`  | `argsOffset` |
-| `top - 9`  | `argsLength` |
-| `top - 10` | `retOffset`  |
-| `top - 11` | `retLength`  |
-
-The signature (`yParity`, `r`, `s`) arguments must be computed from `secp256k1(keccak256(type || abi.encode(invoker, extra)))`.
-
-The arguments are defined to be:
-
- - `type: uint8` - [EIP-2718](./eip-2718.md) transaction type (always `SPONSORED_TYPE`);
- - `invoker: address` - the address of the invoker contract;
- - `extra: uint256` - extra data, which can be used in the invoker to implement replay protection;
- - `callee: address` - address of the callee;
- - `gas: uint256` - exact gas limit which must be provided with the call into `CALLFROM`;
- - `value: uint256` - exact amount of Ether in wei to be received by the callee;
- - `sponsee: address` - address of the sponsee;
- - `argsOffset: uint256`, `argsLength: uint256` - region of memory used as the calldata for the call into the callee;
- - `retOffset: uint256`, `retLength: uint256` - region of memory filled with the return data from the call into the callee; and
- - `yParity: uint8`, `r: bytes32`, `s: bytes32` - signature for the package.
-
-#### Outputs
-
-`CALLFROM` pushes the following two values onto the stack:
+#### Behavior
 
-| Stack      | Value     |
-| ---------- | --------- |
-| `top - 0`  | `valid`   |
-| `top - 1`  | `success` |
+The arguments (`yParity`, `r`, `s`) are interpreted as an ECDSA signature of the form `secp256k1(keccak256(TYPE || abi.encode(invoker, commit)))`, where:
```

Comment:
```suggestion
The arguments (`yParity`, `r`, `s`) are interpreted as an ECDSA signature on the secp256k1 curve over the message `keccak256(TYPE || rlp([invoker, commit]))`, where:
```

Reply:
At minimum, I'd like to use rlp(...) instead of abi.encode(...).

Reply:
or by hand >:(

Reply:
Changed it to specifying it by hand.